libraries/base/Data/Monoid.hs

```diff
@@ -139,7 +152,49 @@ instance Semigroup (Last a) where
 instance Monoid (Last a) where
         mempty = Last Nothing
 
+-- | This data type witnesses the lifting of a monoid into an
+-- applicative pointwise.
```

Comment:
Make sure to put a `@since 4.12.0.0` annotation here too. (This might be the only annotation folks see when searching for this with Hoogle.)